### PR TITLE
Fix client_death forward called on kill command sent by a spectator

### PR DESCRIPTION
**Pull request by WPMGPRoSToTeMa**
This bug can be tested with [Ultimate Gore plugin](https://forums.alliedmods.net/showthread.php?p=19346).

### CODE REVIEW

File: modules/cstrike/csx/meta_api.cpp [hunk 1]
```diff
@@ -125,17 +125,25 @@ const char* get_localinfo( const char* name , const char* def = 0 )
 	return b;
 }
 
-void ClientKill_Post(edict_t *pEntity)
+static bool ClientKill_wasAlive;
```
Comment (souvikdas95) [inline modules/cstrike/csx/meta_api.cpp]: Why use static? Isn't it global already?

Comment (WPMGPRoSToTeMa) [inline modules/cstrike/csx/meta_api.cpp]: For file locality

**Comment by Arkshine** [comment]
If I understand well, the bug is because of the silly engine which doesn't check `deadflag` in `pfnClientKill` and because a spectator is given 1 HP by the game, `pfnClientKill` is unnecessarily fired, then since csx module doesn't check for spectator, `client_death` is fired as well.

Proposed fix looks fine. What about having something similar to the game `IsObserver` and checking for `pev->iuser1`, wouldn't it be enough?

**Comment by WPMGPRoSToTeMa** [comment]
@Arkshine gamedll sets iuser1 for killed player immediately after death.
https://github.com/s1lentq/ReGameDLL_CS/blob/5044d4ed131d683cd38a3728941723aa352bcbdb/regamedll/dlls/player.cpp#L2160

**Comment by Arkshine** [comment]
Looking at the code, this is not guaranteed to be set right away, so I guess the current fix is fine then. Thanks.

**Comment by Arkshine** [comment]
@WPMGPRoSToTeMa are you alive?

**Comment by WPMGPRoSToTeMa** [comment]
@Arkshine yep.

**Comment by Arkshine** [comment]
Don't you see the review above? It misses a `RETURN_META(MRES_IGNORED);`. If you don't provide it metamod will throw an unnecessary warning, if I'm not wrong.

**Comment by WPMGPRoSToTeMa** [comment]
@Arkshine thanks! What review do you mean? I see only this review.
default

**Comment by Arkshine** [comment]
It looks like I did not understand it was necessary to 'submit the review' once you finished to comment. :P

**Pull request merged**